addons/l10n_nl/models/account_journal.py

```diff
@@ -6,6 +6,19 @@
 class AccountJournal(models.Model):
     _inherit = 'account.journal'
 
+    @api.model
+    def _fill_missing_values(self, vals):
+        super()._fill_missing_values(vals)
+
+        if vals.get('type') != 'purchase':
+            return
+
+        company = self.env['res.company'].browse(vals['company_id'])
+        if company.country_id.code == "NL":
+            cost_of_revenue_account_type = self.env["account.account.type"].search([("name", "=", "Cost of Revenue")])
+            if cost_of_revenue_account_type and len(self.type_control_ids) == 0:
```

Comment:
You are in a hook called in `create` before the `create` is actually done: `self.field` will always be False/empty/0